src/android/java/com/pspdfkit/cordova/PSPDFKitCordovaPlugin.java

```diff
@@ -243,4 +254,8 @@ public void onDocumentExtracted(File documentFile) {
     private void showDocumentForUri(@NonNull Uri uri, @NonNull final PdfActivityConfiguration configuration) {
         PdfActivity.showDocument(cordova.getActivity(), uri, configuration);
     }
+
+    private void showDocumentForUriWithSecret(@NonNull Uri uri, @NonNull final PdfActivityConfiguration configuration) {
```

Comment:
You don't require another method for showing documents with password. Please use the existing `showDocumentForUri()` method, and add a `password` parameter. It is valid to pass `null` as password to open unencrypted documents, which means you can get rid of the `if` branches you currently have in `showDocument()`.